test/plugins/retry.js

```diff
@@ -33,7 +33,7 @@ describe('#db Retry Plugin', function() {
         .put(DBNAME)
         .reply(201, {ok: true});
 
-    var cloudantClient = new Client();
+    var cloudantClient = new Client({ plugins: [] });
```

Comment:
I feel like disabling plugins for a test tagged with `#db` is going to cause issues. I think perhaps this test shouldn't be tagged as such since it seems like a mock oriented thing. My original efforts at tagging were directed by the presence of the server URL, which is perhaps what led me astray here.

As you've been doing test fixes anyway - suggest you validate and remove the tag if necessary in your test fix PR.

Reply:
The default behaviour has always been to add the `cookieauth` plugin if no other plugins are specified in the client configuration. This happens in lots of places throughout the tests as we instantiate a client without configuration, i.e. `new Client()`. The problem is there are no mocks for the `POST /_session` calls that happen. This wasn't a problem before since the `cookieauth` plugin would silently swallow the assertion errors thrown by nock.

The alternative fix would be to add in the mocks. I'm happy to do this if you'd prefer. I just thought it makes the test needlessly more complicated.

Reply:
Sorry if I wasn't clear.
I'm fine with this `plugins: []` removing the `cookieauth` plugin for tests that run _only_ as mocks. Which I think is the case in `retry.js`. The aside problem is that this is tagged `#db` which effectively means run against a real DB (wrong for `retry.js` I think), but something that should be addressed in the follow-up changes to #395